### ISSUE TITLE
[DetailsList] Focus not associated with the correct item when navigating with keyboard

### ISSUE THREAD

**Issue by rhublard** [OfficeDev/office-ui-fabric-react#9176]
### Environment Information 
 
- __Package version(s)__: 6.185.0
- __Browser and OS versions__: Happens in Chrome, Edge & Firefox

### Please provide a reproduction of the bug in a codepen:

[https://codesandbox.io/s/focusdetailslist-mo6o7865xy](https://codesandbox.io/s/focusdetailslist-mo6o7865xy)

#### Actual behavior:

When navigating through the list using arrow keys. if the user click on a header and goes back to the rows using `down key`, the first item is selected and the previously selected item is focused. The navigation resume normally to the previously selected item after this.

Everything is fine when using `tab`.
 
We also noticed than when setting SelectionMode to single (as in the codepen), if the previously selected item was the first row, it is unselected when navigating back to it. Is it the intended behavior?

#### Expected behavior:
 
The item focused and the selected item should be the same.

Thanks

**Comment by kenotron** [comment]
Thanks for the sandbox! Am taking a look here.

**Comment by kenotron** [comment]
This was a scenario that we didn't plan for properly from #5966 when there already was selection prior to selecting header and hitting the down key. 

@natalieethell for visibility - would you mind looking at my PR for this?

**Comment by rhublard** [comment]
Thank you for the quick answer/fix ongoing!

**Comment by tipantaz** [comment]
Hello! I have been encountering the same issue but no necessarily by clicking.

I can also reproduce it in your example.
https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist

Version:  "office-ui-fabric-react": "7.5.0"
Chrome,Edge and windows. 

Actual behavior:
Use tab to get to the list header (for example "Name"), use arrow down to get to the first row and press arrow key down one more time to get to the second.

Second one is selected but first one still has focus.

Expected behavior:
Either focus following selection, or no focus.
